### ISSUE TITLE
diff is broken on changes at the end of the file

### ISSUE THREAD

**Issue by radkujawa** [psf/black#1414]
**Description**
diff generated on a file with changes just before the end is broken (cannot be applied with patch command).

**To Reproduce**
1. Take file `test_file.py`:
```
class AClass:
    def a_method(self):

        self.do_something({'key1': 'val1',
                           'key2': 'val2'})
```
2. Run `black --quiet --diff test_file.py > changes.diff`
3. Run `patch -p0 < changes.diff`
4. Get `1 out of 1 hunk FAILED`

**Expected behavior**
expect `patch` command from point 4. to apply the patch

**Environment:**
```
$ pip show black
Name: black
Version: 19.10b0
Summary: The uncompromising code formatter.
Home-page: https://github.com/psf/black
Author: Łukasz Langa
Author-email: lukasz@langa.pl
License: MIT
Location: /home/radek/.virtualenvs/sf/lib/python3.7/site-packages
Requires: attrs, regex, appdirs, toml, typed-ast, pathspec, click
Required-by: 
```

```
$ python --version
Python 3.7.6
```

```
$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="20.04 LTS (Focal Fossa)"
```

**Does this bug also happen on master?**
no (hard to say for sure, looks like latest build is a bit unstable)

**Comment by radkujawa** [comment]
sorry for bothering you, looks like this is fixed on latest build, maybe close it when fix enters stable version?

**Comment by ichard26** [comment]
Can confirm this is fixed in `20.8b1`:

```console
(black) ichard26@acer-ubuntu:~/programming/oss/black$ cat temp.py
class AClass:
    def a_method(self):

        self.do_something({'key1': 'val1',
                           'key2': 'val2'})
(black) ichard26@acer-ubuntu:~/programming/oss/black$ black --quiet --diff temp.py > changes.diff
(black) ichard26@acer-ubuntu:~/programming/oss/black$ patch -p0 < changes.diff
patching file temp.py
``` 

<details>

<summary><b>Environment</b></summary>

* Black version: `20.8b1`
* Python version: `CPython 3.8.5`
* OS version: `Ubuntu 20.04.01 LTS`

</details>

**Comment by radkujawa** [comment]
thank you very much!
